test/Algebra/Graph/Test/Arbitrary.hs

```diff
@@ -74,6 +75,20 @@ instance (Ord a, Arbitrary a) => Arbitrary (AAM.AdjacencyMap a) where
           let edges = AAM.edgeList g
           in [ AAM.removeEdge x y g | (x, y) <- edges ]
 
+-- An Arbitrary instance for Acyclic.Labelled.AdjacencyMap
+instance (Ord a, Arbitrary a, Eq e, Monoid e, Arbitrary e) => Arbitrary (ALAM.AdjacencyMap e a) where
+    arbitrary = ALAM.toAcyclicOrd <$> arbitrary
```

Comment:
This is not a very good `Arbitrary` instance: it can only produce graphs where edges go from "lower" to "higher" vertices.

Reply:
With the functions that we have now since we need to produce Acyclic Labelled graph, I think this is the best we can do. This is similar to the `Arbitrary` instance for `Acyclic.AdjacencyMap`.
Having functions like `shrink` will probably improve the `Arbitrary` instance.

Reply:
Here is what we could do:
1) Generate a random `AdjacencyMap` via `Ord`, like you do.
2) Generate a random shuffle of the sequence `[1..n]`.
3) Use `AM.gmap` to remap all vertices according to the shuffle.

Alas, it looks like we can't make it using our current API, since we won't be able to apply `gmap`. 

What about adding `swapVertices :: a -> a -> AdjacencyMap a -> AdjacencyMap a` to the acyclic API? It seems to be quite safe, and then we could use it when computing a random vertex permutation :)

Reply:
Yes, I agree. `swapVertices` will give more freedom to the graphs.